python/ray/_private/runtime_env/working_dir.py

```diff
@@ -420,12 +420,24 @@ def setup_working_dir(runtime_env: dict,
     # TODO(edoakes): we should be able to remove this by refactoring the
     # working_dir setup code into a class instead of using global vars.
     global _logger, PKG_DIR
-    prev_logger = _logger
-    prev_pkg_dir = PKG_DIR
-    _logger = logger
-    PKG_DIR = context.session_dir
-
-    context.working_dir = ensure_runtime_env_setup(runtime_env["uris"])
-
-    PKG_DIR = prev_pkg_dir
-    _logger = prev_logger
+    if logger:
+        prev_logger = _logger
+        _logger = logger
+    if context.resources_dir:
+        prev_pkg_dir = PKG_DIR
+        PKG_DIR = context.resources_dir
+
+    working_dir = ensure_runtime_env_setup(runtime_env["uris"])
+    context.command_prefix += [f"cd {working_dir}"]
+
+    # Insert the working_dir as the first entry in PYTHONPATH. This is
+    # compatible with users providing their own PYTHONPATH in env_vars.
+    python_path = working_dir
+    if "PYTHONPATH" in context.env_vars:
+        python_path += os.pathsep + runtime_env["PYTHONPATH"]
+    context.env_vars["PYTHONPATH"] = python_path
```

Comment:
This part is a little confusing, should the if statement say `if runtime_env.get("env_vars") and "PYTHONPATH" in runtime_env["env_vars"]`?  And/or should `runtime_env["PYTHONPATH"]` be replaced by `runtime_env["env_vars"]["PYTHONPATH"]`?

Reply:
Ah maybe the first if statement is okay, if we're assuming that context.env_vars has already been updated with the runtime env's env vars.

Reply:
Ah yes, I had intended to initialize `context.env_vars` with the env vars specified in runtime_env. I agree that it's a bit confusing that we're getting it from two places though. My reasoning for this was to avoid having implementations for different options change the runtime_env globally.

Reply:
The latest env vars fix looks good, but should we still replace `runtime_env["PYTHONPATH"]` with `context.env_vars["PYTHONPATH"]`?

Reply:
Ah, thanks!!